File: source/client/web/invoice-book-web/public_html/js/aatl-ib/service/sales-item-service.js
/* 
 * Copyright (c) 2019 Absolute Apogee Technologies Ltd. All rights reserved.
 * 
 * =============================================================================
 * Revision History:
 * Date         Author          Detail
 * -----------  --------------  ------------------------------------------------
 * 2019-Feb-16  GShokar         Created
 * =============================================================================
 */


"use strict";

$aatl_ib.SalesItemService = {

    itemTypeList: [],
    serviceItems: function (callback) {
        $aatl_ib.ApiService.get("sales-item/service-items",
                {},
                function (res, err) {
                    if (err) {
                        callback([], "Service item list request failed: " + err);

                    } else if (res.status === "failure") {
                        callback([], res.message);
                    } else if (res.status === "success") {
                        callback(JSON.parse(res.data));
                    } else {
                        callback([], "Invalid response from server")
                    }
                });
    },
    itemsByItemType: function (itemType, callback) {
        $aatl_ib.ApiService.get("sales-item/items-by-item-type",
                {itemType: itemType},
                function (res, err) {
                    if (err) {
                        callback([], {messages: ["Item list by item type request failed: " + err]});

                    } else if (res.status === "failure") {
                        callback([], res.message);
                    } else if (res.status === "success") {
                        callback(JSON.parse(res.data));
                    } else {
                        callback([], "Invalid response from server")
                    }
                });
    },
       itemTypes: function (callback) {

        if ($aatl_ib.SalesItemService.itemTypeList.length > 0) {
            callback($aatl_ib.SalesItemService.itemTypeList);
        } else {
            $aatl_ib.ApiService.get("sales-item/item-types",
                    {},
                    function (res, err) {
                        if (err) {
                            callback([], "Item types list request failed: " + err);

                        } else if (res.status === "failure") {
                            callback([], res.message);
                        } else if (res.status === "success") {
                            $aatl_ib.SalesItemService.itemTypeList = JSON.parse(res.data);
                            callback($aatl_ib.SalesItemService.itemTypeList);
                        } else {
                            callback([], "Invalid response from server");
                        }
                    });
        }
    }
};

File: source/bl/invoice-book-service/src/main/java/ca/aatl/app/invoicebook/bl/ejb/SalesItemService.java
/*
 * Copyright (c) 2019 Absolute Apogee Technologies Ltd. All rights reserved.
 * 
 * =============================================================================
 * Revision History:
 * Date         Author          Detail
 * -----------  --------------  ------------------------------------------------
 * 2019-Feb-16  GShokar         Created
 * =============================================================================
 */
package ca.aatl.app.invoicebook.bl.ejb;

import ca.aatl.app.invoicebook.data.jpa.dao.SalesItemDao;
import ca.aatl.app.invoicebook.data.jpa.entity.SalesItem;
import ca.aatl.app.invoicebook.data.jpa.entity.SalesItemType;
import java.util.List;
import javax.ejb.EJB;
import javax.ejb.LocalBean;
import javax.ejb.Stateless;

/**
 *
 * @author GShokar
 */

@Stateless
@LocalBean
public class SalesItemService {

    @EJB
    SalesItemDao dao;
    
    public List<SalesItem> serviceItems() throws Exception{
        return dao.listByItemType("Service");
    }

    public SalesItem find(String code) throws Exception{
        return dao.find(code);
    }

    public List<SalesItemType> itemTypes() throws Exception{
        return dao.itemTypes();
    }
    
}
